Metabase downgrade issue - v0.50.7 to v0.49.x

### Describe the bug

Not able to downgrade metabase from v0.50.7 to v0.49.x

**Cluster configuration:**

ECS cluster with 2 tasks.
Task CPU: 3800 units (3.711 vCPU)
Task memory: 3700 MiB (3.613 GB)

**Steps that I followed for downgrading** (followed instructions from [here](https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase#rolling-back-an-upgrade)):

- Stopped the container running v0.50.7 metabase instance

- Ran the following on the EC2 instance directly:
```
# docker run --rm metabase/metabase:v0.50.7 "migrate down"
Warning: environ value jdk-11.0.23+9 for key :java-version has been overwritten with 11.0.23
2024-06-26 18:59:08,086 INFO metabase.util :: Maximum memory available to JVM: 1.9 GB2024-06-26 18:59:10,579 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html2024-06-26 18:59:11,325 WARN db.env :: WARNING: Using Metabase with an H2 application database is not recommended for production deployments. For productiondeployments, we highly recommend using Postgres, MySQL, or MariaDB instead. If you decide to continue to use H2, please be sure to back up the database fileregularly. For more information, see https://metabase.com/docs/latest/operations-guide/migrating-from-h2.html
2024-06-26 18:59:15,823 INFO driver.impl :: Registered abstract driver :sql  🚚2024-06-26 18:59:15,829 INFO driver.impl :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
2024-06-26 18:59:15,835 INFO metabase.util :: Load driver :sql-jdbc took 33.1 ms
2024-06-26 18:59:15,836 INFO driver.impl :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
2024-06-26 18:59:15,977 INFO driver.impl :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
2024-06-26 18:59:16,017 INFO driver.impl :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
2024-06-26 18:59:17,665 INFO metabase.core ::
Metabase v0.50.7 (431cd8f)

Copyright © 2024 Metabase, Inc.
Metabase Enterprise Edition extensions are NOT PRESENT.2024-06-26 18:59:17,956 INFO db.setup :: Setting up Liquibase...
2024-06-26 18:59:18,695 INFO db.liquibase :: Updating liquibase table to reflect consolidated changeset filenames
2024-06-26 18:59:18,707 INFO db.liquibase :: No migration lock found.
2024-06-26 18:59:18,707 INFO db.liquibase :: Migration lock acquired.
2024-06-26 18:59:18,713 INFO db.setup :: Liquibase is ready.2024-06-26 18:59:18,717 INFO db.liquibase :: No migration lock found.
2024-06-26 18:59:18,718 INFO db.liquibase :: Migration lock acquired.2024-06-26 18:59:18,719 INFO db.liquibase :: Rolling back app database schema to version 49
#
```

- Tried spinning up a container using v0.49.12, logs attached
[Downgrade_logs.csv](https://github.com/user-attachments/files/15994385/Downgrade_logs.csv)



### To Reproduce

Check Describe the bug section

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
Diagnostic info:

{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "6.1.91-99.172.amzn2023.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-06-25",
      "tag": "v0.50.7",
      "hash": "431cd8f"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```


### Severity

Urgent. A lot of our clients are getting affected due to frequent timeouts in v0.50.1, and it's becoming a real problem for them.

### Additional context

_No response_

Tried to downgrade it too, but receive an error:
`Metabase v0.50.7 (431cd8f)

Copyright © 2024 Metabase, Inc.

Metabase Enterprise Edition extensions are NOT PRESENT.
2024-06-27 14:12:54,588 INFO db.setup :: Setting up Liquibase...
2024-06-27 14:12:55,485 INFO db.setup :: Liquibase is ready.
2024-06-27 14:12:55,529 INFO db.liquibase :: No migration lock found.
2024-06-27 14:12:55,529 INFO db.liquibase :: Migration lock acquired.
2024-06-27 14:12:55,536 INFO db.liquibase :: Rolling back app database schema to version 49
2024-06-27 14:12:56,031 INFO db.custom-migrations :: No rollback for:  CreateSampleContent
2024-06-27 14:12:56,120 INFO db.custom-migrations :: No rollback for:  MigrateStackedAreaBarComboDisplaySettings
liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.RollbackFailedException: liquibase.exception.DatabaseException: ERROR: column "duration" of relation "task_history" contains null values [Failed SQL: (0) ALTER TABLE "public"."task_history" ALTER COLUMN  "duration" SET NOT NULL]
	at liquibase.command.CommandScope.execute(CommandScope.java:253)
	at liquibase.Liquibase.lambda$rollback$7(Liquibase.java:579)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.Liquibase.runInScope(Liquibase.java:1419)
	at liquibase.Liquibase.rollback(Liquibase.java:568)
	at liquibase.Liquibase.rollback(Liquibase.java:551)
	at liquibase.Liquibase.rollback(Liquibase.java:542)
	at metabase.db.liquibase$rollback_major_version$fn__44519.invoke(liquibase.clj:485)
	at metabase.db.liquibase$run_in_scope_locked$reify__44433.run(liquibase.clj:324)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at metabase.db.liquibase$run_in_scope_locked.invokeStatic(liquibase.clj:317)
	at metabase.db.liquibase$run_in_scope_locked.invoke(liquibase.clj:300)
	at metabase.db.liquibase$rollback_major_version.invokeStatic(liquibase.clj:478)
	at metabase.db.liquibase$rollback_major_version.invoke(liquibase.clj:465)
	at metabase.db.liquibase$rollback_major_version.invokeStatic(liquibase.clj:470)
	at metabase.db.liquibase$rollback_major_version.invoke(liquibase.clj:465)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:671)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.db.setup$migrate_BANG_$fn__53364.invoke(setup.clj:86)
	at metabase.db.liquibase$do_with_liquibase$f_STAR___44374.invoke(liquibase.clj:139)
	at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:142)
	at metabase.db.liquibase$do_with_liquibase.invoke(liquibase.clj:130)
	at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:73)
	at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:55)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.cmd.migrate$migrate_BANG_.invokeStatic(migrate.clj:8)
	at metabase.cmd.migrate$migrate_BANG_.invoke(migrate.clj:5)
	at clojure.lang.Var.invoke(Var.java:384)
	at metabase.cmd$migrate.invokeStatic(cmd.clj:66)
	at metabase.cmd$migrate.invoke(cmd.clj:62)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.cmd$run_cmd$fn__106571.invoke(cmd.clj:301)
	at metabase.cmd$run_cmd.invokeStatic(cmd.clj:300)
	at metabase.cmd$run_cmd.invoke(cmd.clj:290)
	at clojure.lang.Var.invoke(Var.java:388)
	at metabase.core$run_cmd.invokeStatic(core.clj:192)
	at metabase.core$run_cmd.invoke(core.clj:190)
	at metabase.core$entrypoint.invokeStatic(core.clj:214)
	at metabase.core$entrypoint.doInvoke(core.clj:209)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at metabase.bootstrap.main(Unknown Source)
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.RollbackFailedException: liquibase.exception.DatabaseException: ERROR: column "duration" of relation "task_history" contains null values [Failed SQL: (0) ALTER TABLE "public"."task_history" ALTER COLUMN  "duration" SET NOT NULL]
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:151)
	at liquibase.command.core.AbstractRollbackCommandStep.doRollback(AbstractRollbackCommandStep.java:112)
	at liquibase.command.core.AbstractRollbackCommandStep.doRollback(AbstractRollbackCommandStep.java:82)
	at liquibase.command.core.RollbackCountCommandStep.lambda$run$0(RollbackCountCommandStep.java:48)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.command.core.RollbackCountCommandStep.run(RollbackCountCommandStep.java:48)
	at liquibase.command.CommandScope.execute(CommandScope.java:217)
	... 57 more
Caused by: liquibase.exception.RollbackFailedException: liquibase.exception.DatabaseException: ERROR: column "duration" of relation "task_history" contains null values [Failed SQL: (0) ALTER TABLE "public"."task_history" ALTER COLUMN  "duration" SET NOT NULL]
	at liquibase.changelog.ChangeSet.rollback(ChangeSet.java:956)
	at liquibase.changelog.visitor.RollbackVisitor.visit(RollbackVisitor.java:63)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$0(ChangeLogIterator.java:133)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:122)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.Scope.child(Scope.java:252)
	at liquibase.Scope.child(Scope.java:256)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:89)
	... 66 more
Caused by: liquibase.exception.DatabaseException: ERROR: column "duration" of relation "task_history" contains null values [Failed SQL: (0) ALTER TABLE "public"."task_history" ALTER COLUMN  "duration" SET NOT NULL]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:470)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:179)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1303)
	at liquibase.database.AbstractJdbcDatabase.executeRollbackStatements(AbstractJdbcDatabase.java:1327)
	at liquibase.database.AbstractJdbcDatabase.executeRollbackStatements(AbstractJdbcDatabase.java:1333)
	at liquibase.changelog.ChangeSet.rollback(ChangeSet.java:934)
	... 80 more
Caused by: org.postgresql.util.PSQLException: ERROR: column "duration" of relation "task_history" contains null values
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2725)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2412)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:502)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:341)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:326)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:302)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:297)
	at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:464)
	... 86 more
Command failed with exception: liquibase.exception.LiquibaseException: liquibase.exception.RollbackFailedException: liquibase.exception.DatabaseException: ERROR: column "duration" of relation "task_history" contains null values [Failed SQL: (0) ALTER TABLE "public"."task_history" ALTER COLUMN  "duration" SET NOT NULL]`

@noahmoss Tried that. Getting the following error:

```
[root]# docker run  \
  -e "MB_DB_TYPE= postgres" \
  -e "MB_DB_DBNAME= postgres" \
  -e "MB_DB_PORT=5432" \
  -e "MB_DB_USER= postgres" \
  -e "MB_DB_PASS=<pass>” \
  -e "MB_DB_HOST=<host>” \
   --rm metabase/metabase:v0.50.7 "migrate down"

Warning: environ value jdk-11.0.23+9 for key :java-version has been overwritten with 11.0.23
2024-06-27 17:30:01,728 INFO metabase.util :: Maximum memory available to JVM: 1.9 GB
2024-06-27 17:30:04,242 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:619)
        at metabase.db.connection$loading__6789__auto____42274.invoke(connection.clj:1)
        at metabase.db.connection__init.load(Unknown Source)
        at metabase.db.connection__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:703)
        at metabase.db$loading__6789__auto____53409.invoke(db.clj:1)
        at metabase.db__init.load(Unknown Source)
        at metabase.db__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:2088)
        at metabase.models.serialization$loading__6789__auto____53610.invoke(serialization.clj:1)
        at metabase.models.serialization__init.load(Unknown Source)
        at metabase.models.serialization__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:3894)
        at metabase.models.setting$loading__6789__auto____54412.invoke(setting.clj:1)
        at metabase.models.setting__init.load(Unknown Source)
        at metabase.models.setting__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:930)
        at metabase.analytics.prometheus$loading__6789__auto____70244.invoke(prometheus.clj:1)
        at metabase.analytics.prometheus__init.load(Unknown Source)
        at metabase.analytics.prometheus__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.invoke(RestFn.java:3894)
        at metabase.core$loading__6789__auto____107972.invoke(core.clj:1)
        at metabase.core__init.load(Unknown Source)
        at metabase.core__init.<clinit>(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at clojure.lang.RT.classForName(RT.java:2209)
        at clojure.lang.RT.classForName(RT.java:2218)
        at clojure.lang.RT.loadClassForName(RT.java:2237)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6908.invoke(core.clj:6162)
        at clojure.core$load.invokeStatic(core.clj:6161)
        at clojure.core$load.doInvoke(core.clj:6145)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5934)
        at clojure.core$load_one.invoke(core.clj:5929)
        at clojure.core$load_lib$fn__6850.invoke(core.clj:5976)
        at clojure.core$load_lib.invokeStatic(core.clj:5975)
        at clojure.core$load_lib.doInvoke(core.clj:5954)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$load_libs.invokeStatic(core.clj:6017)
        at clojure.core$load_libs.doInvoke(core.clj:6001)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$require.invokeStatic(core.clj:6039)
        at clojure.core$require.doInvoke(core.clj:6039)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$serialized_require.invokeStatic(core.clj:6115)
        at clojure.core$requiring_resolve.invokeStatic(core.clj:6124)
        at clojure.core$requiring_resolve.invoke(core.clj:6118)
        at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
        at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at metabase.bootstrap.main(Unknown Source)
Caused by: java.lang.IllegalArgumentException: No method in multimethod 'env-defaults' for dispatch value: : postgres
        at clojure.lang.MultiFn.getFn(MultiFn.java:156)
        at clojure.lang.MultiFn.invoke(MultiFn.java:229)
        at metabase.db.env$env_STAR_.invokeStatic(env.clj:132)
        at metabase.db.env$env_STAR_.invoke(env.clj:117)
        at metabase.db.env__init.load(Unknown Source)
        at metabase.db.env__init.<clinit>(Unknown Source)
        ... 201 more
```

--- a/dev/src/dev/migrate.clj
+++ b/dev/src/dev/migrate.clj
@@ -42,7 +42,7 @@
                       :where  [:> :orderexecuted {:select   [:orderexecuted]
                                                   :from     [:databasechangelog]
                                                   :where    [:like :id (format "%s%%" id)]
-                                                  :order-by [:orderexecuted :desc]
+                                                  :order-by [[:orderexecuted :desc]]
                                                   :limit    1}]
                       :limit 1})
        :count

--- a/resources/migrations/001_update_migrations.yaml
+++ b/resources/migrations/001_update_migrations.yaml
@@ -7564,6 +7564,7 @@ databaseChangeLog:
             tableName: task_history
             columnDataType: ${timestamp_type}
             columnName: ended_at
+      rollback: # we can't reliably add back the constraint while downgrading
 
   - changeSet:
       id: v50.2024-05-08T09:00:04
@@ -7574,6 +7575,7 @@ databaseChangeLog:
             tableName: task_history
             columnDataType: int
             columnName: duration
+      rollback: # we can't reliably add back the constraint while downgrading
 
   - changeSet:
       id: v50.2024-05-08T09:00:05

